File: programmers/k진수에서 소수 개수 구하기/Programmers92335_2.java
package com.example.algorithm;

import java.util.Arrays;

public class Programmers92335_2 {
    public int solution(int n, int k) {
        String kNumber = toKNumber(n, k);
        String[] strNumbers = kNumber.split("0");
        return (int) Arrays.stream(strNumbers)
                .filter(s -> !s.isEmpty() && isPrimeNumber(Long.parseLong(s)))
                .count();
    }

    private String toKNumber(int n, int k) {
        StringBuilder sb = new StringBuilder();
        while (n > 0) {
            sb.insert(0, n % k);
            n /= k;
        }
        return sb.toString();
    }

    private Boolean isPrimeNumber(long n) {
        int sqrt = (int) Math.sqrt(n);
        if (n == 1) {
            return false;
        }
        for (int i = 2; i <= sqrt; i++) {
            if (n % i == 0) {
                return false;
            }
        }
        return true;
    }
}

File: programmers/k진수에서 소수 개수 구하기/Programmers92335_2.kt
package com.example

import kotlin.math.sqrt

class Programmers92335_2 {
    fun solution(n: Int, k: Int): Int {
        val kNumber = convertToK(n, k)
        val numbers = kNumber
            .split("0")
            .filter { it != "" }
            .map { it.toLong() }
            .filter { isPrimeNumber(it) }
        return numbers.size
    }

    private fun convertToK(n: Int, k: Int): String {
        var num = n
        val sb = StringBuilder()
        while (num > 0) {
            sb.insert(0, num % k)
            num /= k
        }
        return sb.toString()
    }

    private fun isPrimeNumber(n: Long): Boolean {
        if (n == 1L) {
            return false
        }
        val sqrt = sqrt(n.toDouble()).toInt()
        for (i in 2..sqrt) {
            if (n % i == 0L) {
                return false
            }
        }
        return true
    }
}
